src/components/SignIn.js

```diff
@@ -116,15 +151,42 @@ class SignIn extends Component {
     }
   };
 
+  handleOnChangePassword = (e) => {
+    const { error } = this.state;
+    const password = e.target.value;
+    this.setState({ password });
+    if (error) {
+      this.setState({ passwordError: passwordValidator(password) });
+    }
+  };
+
   handleKeypress = (e) => {
+    const { signInMethod } = this.state;
+    // signInMethod email when true
     // sign in by pressing the enter key
     if (e.key === 'Enter') {
-      this.handleSignIn();
+      if (signInMethod) {
+        this.handleSignIn();
+      } else {
+        this.handlePasswordSignIn();
+      }
+    }
+  };
+
+  handleSignInMethod = () => {
+    const { signInMethod } = this.state;
+    // signInMethod email when true
+    if (signInMethod) {
+      this.setState({ signInMethod: false });
+    } else {
+      this.setState({ signInMethod: true });
     }
```

Comment:
`this.setState({signInMethod: !signInMethod})`

Reply:
Actually I don't really get this variable. If it is a boolean it should be call `enableSignInMethod`. But actually it's just a variable defining if you login with a password or a mail only, right? (where the value false represents arbitrarily password (or the opposite)). At some point we will have more ways to login (google, switch, qr, etc) so it would be better if you define an enum : LOGIN_METHODS = { PASSWORD: "password", EMAIL: "email" } and `signInMethod` would be set one of these values (and use switch cases).

Reply:
done!